### PR TITLE
fix(permissions): process `URL` in `Deno.FfiPermissionDescriptor.path` for `revoke()` and `request()`

### LINKED ISSUE #17020: bug(permissions): `revoke()` and `request()` don't modify `desc.path` for FFI
Should these lines include `desc.name === "ffi"` like `query()`?

https://github.com/denoland/deno/blob/890b0653104388bfef60ca7ebd0af758caf5f0be/runtime/js/10_permissions.js#L200

https://github.com/denoland/deno/blob/890b0653104388bfef60ca7ebd0af758caf5f0be/runtime/js/10_permissions.js#L219

### PR DESCRIPTION
**Pull request by iuioiua**
Previously, `Deno.permissions.[revoke|request]()` wouldn't correctly process the `path: URL` when `name` was `ffi`. This change fixes that behaviour and adds a new function, `formDescriptor()`, to ensure `URL` arguments are consistently handled across `Deno.permissions.[query|revoke|request]()`.

Fixes #17020.

Is a regression test for this change needed?

### CODE REVIEW

**Comment by iuioiua** [comment]
Where is the best place to write a regression test?

**Comment by bartlomieju** [comment]
> Is a regression test for this change needed?

Yes, it would be good to have a test for it @iuioiua

**Comment by iuioiua** [comment]
Note: this is needed for #17019.

**Comment by petamoriken** [comment]
@iuioiua It seems that the last commit rolled back the deno_third_party submodule.

**Comment by iuioiua** [comment]
Was it updated or rolled back? How do I fix it? I'm not familiar with submodules.

**Comment by bartlomieju** [comment]
> Was it updated or rolled back? How do I fix it? I'm not familiar with submodules.

Pull latest `main`, merge it into this branch, run `git submodule update --init --recursive`, `git add third_party/`, `git commit`.

**Comment by iuioiua** [comment]
Fixed

File: ext/fetch/20_headers.js [hunk 1]
```diff
@@ -405,7 +405,6 @@
 
     [SymbolFor("Deno.privateCustomInspect")](inspect) {
       const headers = {};
-      // deno-lint-ignore prefer-primordials
```
Comment (petamoriken) [inline ext/fetch/20_headers.js:408]: Please restore this line.

File: ext/fetch/21_formdata.js [hunk 2]
```diff
@@ -299,7 +299,6 @@
     const chunks = [];
     const prefix = `--${boundary}\r\nContent-Disposition: form-data; name="`;
 
-    // deno-lint-ignore prefer-primordials
```
Comment (petamoriken) [inline ext/fetch/21_formdata.js:302]: same here

File: ext/web/09_file.js [hunk 3]
```diff
@@ -350,7 +350,6 @@
       const bytes = new Uint8Array(size);
       const partIterator = toIterator(this[_parts]);
       let offset = 0;
-      // deno-lint-ignore prefer-primordials
```
Comment (petamoriken) [inline ext/web/09_file.js:353]: sane here

**Review by petamoriken — CHANGES_REQUESTED** [review]

File: ext/fetch/20_headers.js (hunk 1 above)
Comment (iuioiua) [reply ext/fetch/20_headers.js]: Thanks. Idk why I did that

Comment (iuioiua) [reply ext/fetch/20_headers.js]: Thanks. Idk why I did that

**Comment by lino-levan** [comment]
LGTM

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks for fixing this @iuioiua!

**Pull request merged**